app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -226,7 +233,72 @@ enum SectionHeaderPlacement {
 const SECTION_HEADER_PLACEMENT: SectionHeaderPlacement = SectionHeaderPlacement::Bottom;
 
 
+
+// ========================
 // === Shape Definition ===
+// ========================
+
+
+// === Selection ===
+
+/// A shape of a selection box. It is used as a mask to show only specific parts of the selection
+/// layers. See [component_group](component_group#selection) module documentation to learn how this
+/// mask is used.
+///
+/// The selection `mask` is a rectangle with rounded corners. The final shape is an intersection of
+/// the `mask` and the `area` shape that covers the entire component list panel and has the same
+/// shape. This is needed to ensure that the selection box never leaves the borders of the
+/// component list panel. Instead of using `set_position` API methods of the display object, we
+/// have a `pos` parameter instead. Similarly, we have the `selection_size` parameter for the
```

Comment:
- Please use a grammar checking tool. I was asking about it in previous PRs. Such sentences as `Instead of using `set_position` API methods of the display object, we have a `pos` parameter instead` are simply incorrect.
- Again, no explanation here. There is info that we dont use `set_position` API, but no explanation of why.

Reply:
The whole section is rewritten for clarity.